pkg/pointerdb/pointerdb.go

```diff
@@ -228,3 +231,70 @@ func (s *Server) Delete(ctx context.Context, req *pb.DeleteRequest) (resp *pb.De
 	s.logger.Debug("deleted pointer at path: " + req.GetPath())
 	return &pb.DeleteResponse{}, nil
 }
+
+func (s *Server) iterate(ctx context.Context, req *pb.IterateRequest, queue dr.Queue) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	s.logger.Debug("entering pointerdb iterate")
+	
+	if err = s.validateAuth(req.GetAPIKey()); err != nil {
+		return err
+	}
+	
+	iter := func(it storage.Iterator) error {
+		var item storage.ListItem
+		for ; req.Limit > 0 && it.Next(&item); req.Limit-- {
+			if item.Key == nil {
+				panic("nil key")
+			}
+			pointer := &pb.Pointer{}
+			err = proto.Unmarshal(item.Value, pointer)
+			if err != nil {
+				//TODO better error handling
+				return err
+			}
+			pieces := pointer.Remote.RemotePieces
+			var nodeIDs []dht.NodeID
+			for i, p := range pieces {
+				nodeIDs = append(nodeIDs, kademlia.StringToNodeID(p.NodeId))
+			}
+			missingPieces, err := offlineNodes(ctx, nodeIDs)
+			if err != nil {
+				//TODO better error handling
+				return err
+			}
+			if int32(len(missingPieces)) >= pointer.Remote.Redundancy.RepairThreshold {
+				err = queue.Add(&pb.InjuredSegment{
```

Comment:
Why should this ever error?

Reply:
perhaps adding to redis will error, not sure yet